address more safer cpp failures in WebKit/UIProcess/Cocoa

cdumez opened the pull request:
#### 87d8c1ec3c3acbd65aa444f628629c120e6caeea
<pre>
address more safer cpp failures in WebKit/UIProcess/Cocoa
<a href="https://bugs.webkit.org/show_bug.cgi?id=291249">https://bugs.webkit.org/show_bug.cgi?id=291249</a>

Reviewed by Darin Adler.

* Source/WebKit/SaferCPPExpectations/UnretainedLocalVarsCheckerExpectations:
* Source/WebKit/UIProcess/Cocoa/ModelElementControllerCocoa.mm:
(WebKit::ModelElementController::getCameraForModelElement):
(WebKit::ModelElementController::setCameraForModelElement):
(WebKit::ModelElementController::isPlayingAnimationForModelElement):
(WebKit::ModelElementController::setAnimationIsPlayingForModelElement):
(WebKit::ModelElementController::isLoopingAnimationForModelElement):
(WebKit::ModelElementController::setIsLoopingAnimationForModelElement):
(WebKit::ModelElementController::animationDurationForModelElement):
(WebKit::ModelElementController::animationCurrentTimeForModelElement):
(WebKit::ModelElementController::setAnimationCurrentTimeForModelElement):
(WebKit::ModelElementController::hasAudioForModelElement):
(WebKit::ModelElementController::isMutedForModelElement):
(WebKit::ModelElementController::setIsMutedForModelElement):
* Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
(WebKit::UIDelegate::UIClient::requestStorageAccessConfirm):
(WebKit::UIDelegate::UIClient::focusFromServiceWorker):
(WebKit::UIDelegate::UIClient::didChangeFontAttributes):
(WebKit::UIDelegate::UIClient::callDisplayCapturePermissionDelegate):
(WebKit::UIDelegate::UIClient::decidePolicyForUserMediaPermissionRequest):
(WebKit::UIDelegate::UIClient::decidePolicyForScreenCaptureUnmuting):
(WebKit::UIDelegate::UIClient::queryPermission):
(WebKit::UIDelegate::UIClient::didEnableInspectorBrowserDomain):
(WebKit::UIDelegate::UIClient::didDisableInspectorBrowserDomain):
(WebKit::UIDelegate::UIClient::updateAppBadge):
(WebKit::UIDelegate::UIClient::didAdjustVisibilityWithSelectors):
(WebKit::UIDelegate::UIClient::recentlyAccessedGamepadsForTesting):
(WebKit::UIDelegate::UIClient::stoppedAccessingGamepadsForTesting):
(WebKit::UIDelegate::UIClient::requestPermissionOnXRSessionFeatures):
(WebKit::UIDelegate::UIClient::supportedXRSessionFeatures):
(WebKit::UIDelegate::UIClient::startXRSession):
(WebKit::UIDelegate::UIClient::endXRSession):
* Source/WebKit/UIProcess/Cocoa/WKContactPicker.mm:
(-[WKContactPicker _contactInfoFromCNContact:]):
(-[WKContactPicker _contactsFromJSContacts:]):
* Source/WebKit/UIProcess/Cocoa/WKShareSheet.mm:
(typeIdentifierForFileURL):
(nameForFileURLWithTypeIdentifier):
(placeholderMetadataWithFileURL):
(appendFilesAsShareableURLs):
(-[WKShareSheet presentWithShareDataArray:inRect:]):
(+[WKShareSheet setQuarantineInformationForFilePath:]):
(+[WKShareSheet createTemporarySharingDirectory]):
(+[WKShareSheet createRandomSharingDirectoryForFile:]):
(+[WKShareSheet writeFileToShareableURL:data:temporaryDirectory:]):

Canonical link: <a href="https://commits.webkit.org/293461@main">https://commits.webkit.org/293461@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3ac90ac37e173bfc7cdc8c50f3f30b9570ce5d7d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98985 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18622 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8866 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104114 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/49577 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18917 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27072 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75353 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32481 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101989 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14380 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89394 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55714 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14172 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7369 "Build is in progress. Recent messages:") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/48954 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84101 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7444 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106480 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26082 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/19013 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84316 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26457 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85591 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83819 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28477 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6146 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/19809 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16098 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26035 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/31221 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25855 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29175 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27429 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
```diff
@@ -1278,17 +1278,17 @@ static _WKAutoplayEvent toWKAutoplayEvent(WebCore::AutoplayEvent event)
         return;
 
-    auto privateUIDelegate = (id<WKUIDelegatePrivate>)uiDelegate->m_delegate.get();
+    RetainPtr privateUIDelegate = (id<WKUIDelegatePrivate>)uiDelegate->m_delegate.get();
```
Seems like could use a helper function for this so we don’t have to have these type casts all over this file?

darinadler commented on Source/WebKit/UIProcess/Cocoa/WKShareSheet.mm:
```diff
@@ -94,16 +94,16 @@ static uint64_t sizeForFileURL(NSURL *url)
-    NSDictionary<NSURLResourceKey, id> *itemResources = [url promisedItemResourceValuesForKeys:@[ NSURLHasHiddenExtensionKey, NSURLLocalizedNameKey ] error:nil];
-    if (NSString *localizedName = [itemResources objectForKey:NSURLLocalizedNameKey]) {
+    RetainPtr<NSDictionary<NSURLResourceKey, id>> itemResources = [url promisedItemResourceValuesForKeys:@[ NSURLHasHiddenExtensionKey, NSURLLocalizedNameKey ] error:nil];
+    if (RetainPtr<NSString> localizedName = [itemResources objectForKey:NSURLLocalizedNameKey]) {
```
`checked_objc_cast` here?

darinadler commented on Source/WebKit/UIProcess/Cocoa/WKShareSheet.mm:
```diff
@@ -94,16 +94,16 @@ static uint64_t sizeForFileURL(NSURL *url)
+    if (RetainPtr<NSString> localizedName = [itemResources objectForKey:NSURLLocalizedNameKey]) {
         if (!isFolder) {
-            if (NSNumber *hasHiddenExtension = [itemResources objectForKey:NSURLHasHiddenExtensionKey]) {
+            if (RetainPtr<NSNumber> hasHiddenExtension = [itemResources objectForKey:NSURLHasHiddenExtensionKey]) {
```
`checked_objc_cast` here?

darinadler commented on Source/WebKit/UIProcess/Cocoa/WKShareSheet.mm:
```diff
@@ -352,7 +352,7 @@ - (void)presentWithParameters:(const WebCore::ShareDataWithParsedURL &)data inRe
 - (void)presentWithShareDataArray:(NSArray *)sharingItems inRect:(std::optional<WebCore::FloatRect>)rect
 {
-    WKWebView *webView = _webView.getAutoreleased();
+    RetainPtr webView = _webView.get();
```
```suggestion
    RetainPtr webView = _webView;
```

cdumez replied:
This doesn't build. `_webView` is a `WeakObjCPtr`, not a `RetainPtr`.

cdumez committed:
address more safer cpp failures in WebKit/UIProcess/Cocoa
https://bugs.webkit.org/show_bug.cgi?id=291249

Reviewed by Darin Adler.

* Source/WebKit/SaferCPPExpectations/UnretainedLocalVarsCheckerExpectations:
* Source/WebKit/UIProcess/Cocoa/ModelElementControllerCocoa.mm:
(WebKit::ModelElementController::getCameraForModelElement):
(WebKit::ModelElementController::setCameraForModelElement):
(WebKit::ModelElementController::isPlayingAnimationForModelElement):
(WebKit::ModelElementController::setAnimationIsPlayingForModelElement):
(WebKit::ModelElementController::isLoopingAnimationForModelElement):
(WebKit::ModelElementController::setIsLoopingAnimationForModelElement):
(WebKit::ModelElementController::animationDurationForModelElement):
(WebKit::ModelElementController::animationCurrentTimeForModelElement):
(WebKit::ModelElementController::setAnimationCurrentTimeForModelElement):
(WebKit::ModelElementController::hasAudioForModelElement):
(WebKit::ModelElementController::isMutedForModelElement):
(WebKit::ModelElementController::setIsMutedForModelElement):
* Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
(WebKit::UIDelegate::UIClient::requestStorageAccessConfirm):
(WebKit::UIDelegate::UIClient::focusFromServiceWorker):
(WebKit::UIDelegate::UIClient::didChangeFontAttributes):
(WebKit::UIDelegate::UIClient::callDisplayCapturePermissionDelegate):
(WebKit::UIDelegate::UIClient::decidePolicyForUserMediaPermissionRequest):
(WebKit::UIDelegate::UIClient::decidePolicyForScreenCaptureUnmuting):
(WebKit::UIDelegate::UIClient::queryPermission):
(WebKit::UIDelegate::UIClient::didEnableInspectorBrowserDomain):
(WebKit::UIDelegate::UIClient::didDisableInspectorBrowserDomain):
(WebKit::UIDelegate::UIClient::updateAppBadge):
(WebKit::UIDelegate::UIClient::didAdjustVisibilityWithSelectors):
(WebKit::UIDelegate::UIClient::recentlyAccessedGamepadsForTesting):
(WebKit::UIDelegate::UIClient::stoppedAccessingGamepadsForTesting):
(WebKit::UIDelegate::UIClient::requestPermissionOnXRSessionFeatures):
(WebKit::UIDelegate::UIClient::supportedXRSessionFeatures):
(WebKit::UIDelegate::UIClient::startXRSession):
(WebKit::UIDelegate::UIClient::endXRSession):
* Source/WebKit/UIProcess/Cocoa/WKContactPicker.mm:
(-[WKContactPicker _contactInfoFromCNContact:]):
(-[WKContactPicker _contactsFromJSContacts:]):
* Source/WebKit/UIProcess/Cocoa/WKShareSheet.mm:
(typeIdentifierForFileURL):
(nameForFileURLWithTypeIdentifier):
(placeholderMetadataWithFileURL):
(appendFilesAsShareableURLs):
(-[WKShareSheet presentWithShareDataArray:inRect:]):
(+[WKShareSheet setQuarantineInformationForFilePath:]):
(+[WKShareSheet createTemporarySharingDirectory]):
(+[WKShareSheet createRandomSharingDirectoryForFile:]):
(+[WKShareSheet writeFileToShareableURL:data:temporaryDirectory:]):

Canonical link: https://commits.webkit.org/293461@main
